hoomd/ParticleGroup.cc

```diff
@@ -726,30 +730,42 @@ void ParticleGroup::thermalizeParticleMomenta(Scalar kT, uint64_t timestep)
                                                m_sysdef->getSeed()),
                                    hoomd::Counter(ptag));
 
-        // Generate a random velocity
+        // Generate a random velocity, excluding constituent particles
         Scalar mass = h_vel.data[j].w;
         Scalar sigma = slow::sqrt(kT / mass);
         hoomd::NormalDistribution<Scalar> normal(sigma);
-        h_vel.data[j].x = normal(rng);
-        h_vel.data[j].y = normal(rng);
-        if (n_dimensions > 2)
-            h_vel.data[j].z = normal(rng);
+        // check if particles are constituent particles
+        if (h_tag.data[j] != h_body.data[j] && h_body.data[j] != static_cast<unsigned int>(-1))
```

Comment:
```suggestion
        if (h_tag.data[j] != h_body.data[j] && h_body.data[j] != NO_BODY)
```
Use defined constants in place of hard-coded values.